src/Microsoft.OpenApi/Extensions/EnumExtensions.cs

```diff
@@ -21,6 +23,7 @@ public static class EnumExtensions
         /// <returns>
         /// The attribute of the specified type or null.
         /// </returns>
+        [RequiresUnreferencedCode("GetAttributeOfType is not trim-compatible. Recommended to use native AoT-friendly type-specific overloads of GetDisplayName instead.")]
```

Comment:
If instead of `type.GetMember(enumValue.ToString())` this used `type.GetField(enumValue.ToString())`, it would fall into the case covered by https://github.com/dotnet/runtime/issues/97737. In [.NET 9 the trim warnings](https://github.com/dotnet/runtime/pull/100814/) changed to no longer warn on GetField on a Type guaranteed to be an enum.

If we made that change, until .NET 9, we could suppress the warning saying that the trimmer never trims fields of an enum. For example: https://github.com/dotnet/runtime/blob/ac0eeb015a0266adb06696bb58c21e5bfe6d9b56/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs#L249-L250

cc @MichalStrehovsky

Reply:
I don't expect anyone other than us to have taken a dependency on this. Regardless, removing it would be a source breaking change for the dependents. We should add an obsolete attribute as well so we remove this with v2

Reply:
`type.GetField(enumValue.ToString())` - also pass the binding flags to look for public static members so that we don't accidentally get the "secret" field holding the enum value.

Note that this will probably still warn in 9.0 because I did the `T : enum` handling but not a variable of type `System.Enum` case - that would require extra work. We can try to squeeze it into 9.0 if you think it would be valuable and file a bug on it.

Reply:
> Note that this will probably still warn in 9.0 because I did the T : enum handling but not a variable of type System.Enum case - that would require extra work. We can try to squeeze it into 9.0 if you think it would be valuable and file a bug on it.

I never caught this before. Did you test the [case in the original issue](https://github.com/dotnet/runtime/issues/97737#issue-2109092800)?

```csharp
        private static string? GetEnumName<T>(T value) where T : Enum
        {
            var type = value.GetType();
...
            if(type.GetField(name)?.GetCustomAttribute<EnumMemberAttribute>() is { } attribute)
                return attribute.Value;
```

Note that this isn't using `typeof(T)` but instead `value.GetType()`. See [why this is important to that scenario](https://github.com/dotnet/runtime/issues/97737#issuecomment-1918250306).

That scenario and this current one seem to be the same - variable `Enum foo` => `foo.GetType().GetField("")`